rfc/0052-sdk3-full-text-search.md

```diff
@@ -672,33 +675,29 @@ cluster.searchQuery("search_index_name", query);
 A more complex example showing multiple vector queries and all parameters:
 
 ```java
-// Setting all possible parameters on a VectorQuery.
+// Sending multiple `VectorQuery`s, and setting all possible parameters.
```

Comment:
We support multiple queries through the use of ConjectionQuery and DisjunctionQuery.  Is there a reason behind exposing multiple vector queries through a different mechanism than what we have today?

I guess a related thing that comes to mind is that this API implies that a 'VectorQueries' object "wraps" a search object (ie: that the Vector Query is part of the search being performed).  But the actual API that's being exposed shows that the "query" and "knn" fields are both top-level concepts, which would make me think that the vector querying was meant to be exposed as a search option rather than as a SearchQuery type?  What was the reasoning behind hiding this distinction in the SDK API?

Reply:
> We support multiple queries through the use of ConjectionQuery and DisjunctionQuery. Is there a reason behind exposing multiple vector queries through a different mechanism than what we have today?

Those two already have a defined meaning though - they combine queries inside the "query" field, e.g. the normal non-vector query.
They're also non-trivial and I don't want users to have to understand conjunctions vs disjunctions just to send a vector query.

> the vector querying was meant to be exposed as a search option rather than as a SearchQuery type?

Yes, it would be possible to expose in the options block instead - but that feels weird to me, and also that we're relegating vector search to fairly minor tertiary functionality.  Let's see what SDK PMs think.

Reply:
That's fair.  I guess ultimately the question I was trying to get at is:  _if_ the intent is that it is a 'top level' feature, such as the queries themselves.  Why isn't the FTS API modeled a different way?  Ultimately this is a net-new feature we are adding, and if we presuppose that PM is the one providing guidance on it's external interface, why the incongruence between teams?